### DISCUSSION TITLE
Problem load model using streamlit

### DISCUSSION THREAD

**Discussion by AliceKoh** [pycaret/pycaret#2816] [Q&A]
Hi I create a model using google colab. Successfully save, load and download. However when I load the model in streamlit, there is an error message. Attached is the link to my model .pkl file for your reference.
https://drive.google.com/drive/folders/1398HZU004z6dwscFjJQ16fhxwm1JPquK?usp=sharing

Error Message:
module pycaret.internal.preprocess has no attribute datatype auto infer. 

Please assist.

**Comment by ngupta23** [comment]
Can not access the google drive location.

**Reply by AliceKoh** [reply]
I tried to drag and drop .pkl file but not successful. This is my email alice.python21@gmail.com
you email me. I drop you the file.

**Reply by ngupta23** [reply]
I can not share my email ID here, but you can submit a complete reproducible example with sample data if you want. 
Also, make sure that you are using the latest version of pycaret

`pip install -U --pre pucaret`

**Reply by AliceKoh** [reply]
I try to drop py file also cannot work on github. What is reproducible example ? 
Are you at the pycaret slake? I just sign up. Can I drop the file there?
my pycaret version is 3.0.0rc3. anaconda doesnt allow me to change the version. Is 3 = pre pucaret?

**Reply by AliceKoh** [reply]
It is ok. Someone else assist. It looks like my colab pycaret version and my anaconda pycaret version for streamlit are different thus the model cannot work at streamlit.  I tried on re-do the model using anaconda jupyter and there isnt any error when I load at anaconda same version streamlit. I am going to apply to my apps now....hope it works too.

**Reply by ngupta23** [reply]
> It looks like my colab pycaret version and my anaconda pycaret version for streamlit are different thus the model cannot work at streamlit. I tried on re-do the model using anaconda jupyter and there isnt any error when I load at anaconda same version streamlit. I am going to apply to my apps now....hope it works too.

That is great. Yes, using the same versions is important while reusing the models in different systems.
